app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -1,33 +1,141 @@
+import com.andrew.tasky.agenda.domain.models.EventPhoto
+import com.andrew.tasky.agenda.domain.models.UpsertEventResult
 import com.andrew.tasky.auth.util.getResourceResult
 import com.andrew.tasky.core.Resource
 import com.andrew.tasky.core.UiText
 import javax.inject.Inject
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
-import okio.IOException
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.toRequestBody
+
+typealias LocalPhoto = EventPhoto.Local
 
 class EventRepositoryImpl @Inject constructor(
     private val db: AgendaDatabase,
     private val api: EventApi,
     private val context: Context
 ) : EventRepository {
 
-    override suspend fun upsertEvent(event: AgendaItem.Event): Resource<Unit> {
+    override suspend fun upsertEvent(event: AgendaItem.Event): UpsertEventResult {
+
+        val (compressedEvent, photosDeleted) = compressEvent(event)
+
+        val isEventInDb = db.getEventDao().getEventById(compressedEvent.id) == null
+        val isModifiedCreateEvent = db.getEventDao().getModifiedEventById(compressedEvent.id)
+            ?.modifiedType == ModifiedType.CREATE
 
-        db.getEventDao().upsertEvent(event.toEventEntity())
-        return Resource.Success()
+        db.getEventDao().upsertEvent(compressedEvent.toEventEntity())
+
+        if (isEventInDb || isModifiedCreateEvent) {
+            val result = getResourceResult {
+                api.createEvent(
+                    eventData = MultipartBody.Part
+                        .createFormData(
+                            name = "create_event_request",
+                            value = Json.encodeToString(compressedEvent.toCreateEventRequest())
+                        ),
+                    photoData = compressedEvent.photos.filterIsInstance<LocalPhoto>().mapIndexed {
+                        index, eventPhoto ->
+                        MultipartBody.Part
+                            .createFormData(
+                                name = "photo$index",
+                                filename = eventPhoto.key,
+                                body = eventPhoto.byteArray!!.toRequestBody()
+                            )
+                    }
+                )
+            }
+            return when (result) {
+                is Resource.Error -> {
+                    db.getEventDao().upsertModifiedEvent(
+                        ModifiedEventEntity(
+                            id = compressedEvent.id,
+                            modifiedType = ModifiedType.CREATE
+                        )
+                    )
+                    compressedEvent.photos.filterIsInstance<LocalPhoto>().forEach {
+                        saveLocalPhotoInternally(it)
+                        it.savedInternally = true
+                    }
+                    UpsertEventResult(photosTooBig = photosDeleted, resource = Resource.Error())
```

Comment:
I'm not sure if I understand why the deleted photos are the too big photos here? 🤔